src/client/shadows/dynamicshadows.cpp

```diff
@@ -41,10 +41,13 @@ static v3f quantizeDirection(v3f direction, float step)
 
 void DirectionalLight::createSplitMatrices(const Camera *cam)
 {
-	const float DISTANCE_STEP = BS * 2.0; // 2 meters
+	static const float COS_15_DEG = 0.965926f;
 	v3f newCenter;
-	v3f look = cam->getDirection();
-	look = quantizeDirection(look, M_PI / 12.0); // 15 degrees
+	v3f look = cam->getDirection().normalize();
+	if (look.dotProduct(last_look) >= COS_15_DEG)
```

Comment:
Can you explain how this works? I would have expected `< COS_15_DEG` here.

Reply:
cos is smaller for bigger absolute angles.

Reply:
Right, it only changes shadow map's frustum angle if player's look direction changes more than 15 degrees.
That if statement literally reads as "if angle between current look and last captured look is less than 15 degrees, stick to the last captured value, otherwise (>15 degrees) recapture look direction"

Reply:
D'oh. Makes sense.

Reply:
Perhaps put a comment to that extend here.

Reply:
Done